### PR TITLE
Fix EZP-25203: Pressing <Enter> when creating a role saves the role but don't publish it

### PR DESCRIPTION
**Pull request by glye**
> Fixes https://jira.ez.no/browse/EZP-25203
> Also required: https://github.com/ezsystems/repository-forms/pull/65
> Status: Ready for review

Let the default action call the save role action. Let the save role action both save and publish.

### CODE REVIEW

**Comment by andrerom** [comment]
+1

**Comment by bdunogier** [comment]
Not 100% convinced, but I don't have anything better to suggest. Would need to look deeper into it.

+1 if it works...

File: Form/Processor/RoleFormProcessor.php [hunk 1]
```diff
@@ -56,12 +56,19 @@ public static function getSubscribedEvents()
 
     public function processDefaultAction(FormActionEvent $event)
     {
-        $this->notify('role.notification.draft_saved', [], 'role');
+        // Don't update anything if we just want to cancel the draft.
+        // Don't do the default action when saving is specified, as it's the same action and shouldn't be done twice.
+        if (in_array($event->getClickedButton(), ['removeDraft', 'saveRole'])) {
```
Comment (dpobel) [inline Form/Processor/RoleFormProcessor.php]: wouldn't it be more logic to check if no button was clicked ?

Comment (glye) [inline Form/Processor/RoleFormProcessor.php]: @dpobel Something like this? Ok. Also I can move the explaining comment to the docblock.
```php
if ($event->getClickedButton() === null) {
    $this->processSaveRole($event);
}
```

**Comment by glye** [comment]
Changed the default action to save if no button was clicked. (No functionality change, just clearer code.) @dpobel

**Comment by bdunogier** [comment]
+1

**Pull request merged**